packages/runner-shared/src/container/index.jsx

```diff
@@ -82,18 +79,23 @@ class Container extends Component {
   _automationDisconnected () {
     return <AutomationDisconnected onReload={this.props.eventManager.launchBrowser} />
   }
-}
 
-Container.defaultProps = {
-  eventManager,
-  util,
-}
+  // This two functions, _noSpec anad _checkSpecFile, are only used by the E2E runner.
+  // TODO: remove any runner specific code from this file.
+  _noSpec () {
+    const { NoSpec } = this.props
 
-Container.propTypes = {
-  config: PropTypes.object.isRequired,
-  state: PropTypes.instanceOf(State),
```

Comment:
This was tricky, since the `Container.jsx` is now shared, it's not trivial to import `State` and use it in the propType validation. What we should really do is change this to TypeScript, then we could use a generic type to validate this.

I tried to avoid changing `jsx` to `tsx` for anything but the most simple files in this PR to keep it as small as possible, so I think this is the most practical option. In my experience `propTypes` is not super useful anyway, and our tests are much more thorough and give us enough coverage.